nix/packages/default.nix

```diff
@@ -2,14 +2,33 @@
   perSystem = { pkgs, inputs', system, config, ... }:
     {
       packages = {
+        refresh-available-snapshots-state = pkgs.writeShellApplication {
+          name = "refresh-available-snapshots-state";
+          runtimeInputs = with pkgs; [ awscli2 jq ];
+          text =
+            let
+              # TODO: handle this variable better (e.g. move in a better place, some vars are duplicated in scheduled tasks) 
+              awsEndpoint = "https://pub-b887f41ffaa944ebaae543199d43421c.r2.dev/";
+              outFile = "available-snapshots.json";
+              bucketName = "cardanow";
+            in
+            ''
+              aws s3api list-objects-v2 --bucket ${bucketName} --query "Contents[].{Key:Key, LastModified:LastModified}" | jq '.[] | .Key = "${awsEndpoint}" + .Key' > ${outFile} && aws s3 cp ${outFile} s3://${bucketName}/available-snapshots.json
+            '';
+        };
         cleanup-local-data = pkgs.writeShellApplication {
           name = "cleanup-local-data";
           runtimeInputs = with pkgs; [ bash ];
-          text = ../../bin/cleanup-local-data.sh;
+          text = ''${../../bin/cleanup-local-data.sh} "$@"'';
         };
         cleanup-s3-data = pkgs.writeShellApplication {
           name = "cleanup-local-data";
-          runtimeInputs = with pkgs; [ awscli2 bash jq ];
+          runtimeInputs = with pkgs; [
+            awscli2
+            bash
+            config.packages.refresh-available-snapshots-state
+            jq
+          ];
           text = ''${../../bin/cleanup-s3-data.sh} "$@"'';
```

Comment:
Can't you change this to `text = ../../bin/cleanup-s3-data.sh` as you did for `cleanup-local-data`?

Reply:
What do you mean? I think I have to do this to propagate arguments, cleanup local data has the same thing :thinking: 
https://github.com/mlabs-haskell/cardanow/blob/89233320d727519adda7064cbfa4592c392bd9e5/nix/packages/default.nix#L22

Am I missing something?

Reply:
When you execute `nix run .#package -- foo bar` you are already passing `foo` and `bar` to the script built by building the `package` derivation. You can see it as a poor man's beta reduction :lol:

Just tried this:
```nix
perSystem = {pkgs, ...}: {
  packages.test = pkgs.writeShellScriptBin "test" ''
    echo $@
  '';
};
```

```
$ nix run .#test -- foo bar
foo bar
```

> cleanup local data has the same thing

Probably I expressed myself badly, I meant applying this wherever you can.